firmware/raspi/gui.py

```diff
@@ -0,0 +1,359 @@
+    dim_square = 64
+
+    def __init__(self, parent, board):
+        self.chessboard = board
+        self.parent = parent
+        canvas_width = self.columns * self.dim_square
+        canvas_height = self.rows * self.dim_square
+        self.canvas = tk.Canvas(parent, width=canvas_width,
+                               height=canvas_height, bg="black")
+        self.canvas.pack(padx=8, pady=8)
+        #Adding Frame
+        self.btmfrm = tk.Frame(parent, height=64)
+        self.btmfrm.pack(fill="x", side=tk.BOTTOM)
+
+    def setup_game(self):
+        """Sets up the main game variables and then calls the tkinter mainloop
+        which waits for events
+        """
+        params = init_parameters()
+        mode_of_interaction = params["mode_of_interaction"]
+        print(f"\nRUNNING IN {mode_of_interaction.upper()} MODE...\n")
+ 
+        if params["mode_of_interaction"] == "test":
+            self.game = Game(params["mode_of_interaction"], params["interact_w_arduino"])
+            # TODO: Refactor to handle dispatching moves using the code in `process`.
+            # raise ValueError("Test mode of interaction not yet implemented.")
+        elif params["mode_of_interaction"] == "debug":
+            self.game = Game(params["mode_of_interaction"], params["interact_w_arduino"])
+            # TODO: Implement debug mode of interaction
+            # Should be able to use process with human as both players
+            # raise ValueError("Debug mode of interaction not yet implemented.")
+        elif params["mode_of_interaction"] in ("cli", "otb", "web", "speech"):
+            self.game = Game(params["mode_of_interaction"], params["interact_w_arduino"],
+                                params["human_plays_white_pieces"])
+            # _, human_plays_white_pieces, board, ai_player = params
+
+        self.players = params["players"]
+        if params["human_plays_white_pieces"]:
+            self.turn = 0
+            self.move_first = "gui"
+        else:
+            self.turn = 1
+            self.move_first = "ai"
+          
+        if not self.turn:
+            print("Please make your move on the board.")
+        else:
+            self.game.process(self.players[0])
+            self.chessboard.show(self.game.current_fen())
+            self.draw_board()
+            self.show_move(self.game.last_made_move(), self.game.is_white_turn())
+            self.draw_pieces()
+            self.turn = 1
+        self.parent.after(10, self.update_state)
+        self.parent.mainloop()
+
+    def update_state(self):
+        """Runs simultaneously with the tkinter main loop and checks if a move has been
+        made every 10ms 
+        """
+        made_move = self.game.process(self.players[self.turn])
+        if made_move:
+            self.selected_piece = None
+            self.focused = None
+            self.pieces = {}
+            if (self.turn and self.move_first == "gui") or (not self.turn and self.move_first == "ai"):
```

Comment:
Yes so this if statement will print the arrows on the turn of the opposing player to the human. If the turn is 1/true and the human moved first, then the opposing player moved and we need to show the arrows. If the turn is 0/false and the AI moved first, then the opposing player moved and we need to show the arrows. Since turn will represent the opposing player when it is true and false (depending on which color the human picked), I wrote this if statement to work with the respective conditions

Reply:
Sounds good; can you add a brief comment explaining that and update this condition to be:
```py
if (self.turn and self.human_moves_first) or (not self.turn and not self.human_moves_first):
```